libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -333,9 +338,71 @@ - (SFRestRequest *)requestForSearchResultLayout:(NSString*)objectList {
     return [SFRestRequest requestWithMethod:SFRestMethodGET path:path queryParams:queryParams];
 }
 
-+ (NSString *)getHttpStringFomFromDate:(NSDate *)date {
-    if (date == nil) return nil;
+- (SFRestRequest *) batchRequest:(NSArray<SFRestRequest*>*) requests haltOnError:(BOOL) haltOnError {
+    NSMutableArray *requestsArrayJson = [NSMutableArray new];
+    for (SFRestRequest *request in requests) {
+        NSMutableDictionary<NSString *, id> *requestJson = [NSMutableDictionary new];
+        requestJson[@"method"] = [SFRestRequest httpMethodFromSFRestMethod:request.method];
+        // queryParams belong in url
+        if (request.method == SFRestMethodGET || request.method == SFRestMethodDELETE) {
+            requestJson[@"url"] = [NSString stringWithFormat:@"%@%@", request.path, [self toQueryString:request.queryParams]];
+        }
+        // queryParams belongs in body
+        else {
+            requestJson[@"url"] = request.path;
+            requestJson[@"richInput"] = request.queryParams;
+        }
+        [requestsArrayJson addObject:requestJson];
+    }
+    NSMutableDictionary<NSString *, id> *batchRequestJson = [NSMutableDictionary new];
+    batchRequestJson[@"batchRequests"] = requestsArrayJson;
+    batchRequestJson[@"haltOnError"] = [NSNumber numberWithBool:haltOnError];
+
+    NSString *path = [NSString stringWithFormat:@"/%@/composite/batch", self.apiVersion];
+
+    return [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:batchRequestJson];
+}
+
+- (SFRestRequest *) compositeRequest:(NSArray<SFRestRequest*>*) requests refIds:(NSArray<NSString*>*)refIds allOrNone:(BOOL) allOrNone {
+    NSMutableArray *requestsArrayJson = [NSMutableArray new];
+    for (int i=0; i<requests.count; i++) {
+        SFRestRequest *request = requests[i];
+        NSString *refId = refIds[i];
+        NSMutableDictionary<NSString *, id> *requestJson = [NSMutableDictionary new];
+        requestJson[@"referenceId"] = refId;
+        requestJson[@"method"] = [SFRestRequest httpMethodFromSFRestMethod:request.method];
+        // queryParams belong in url
+        if (request.method == SFRestMethodGET || request.method == SFRestMethodDELETE) {
+            requestJson[@"url"] = [NSString stringWithFormat:@"%@%@%@", request.endpoint, request.path, [self toQueryString:request.queryParams]];
```

Comment:
For composite request the url field contains the end point (but for batch request, it's whatever comes after the end point which is expected to be /services/data)